extensions/AzureCosmosDBMongoDB/AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemory.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.AI;
+using Microsoft.KernelMemory.Diagnostics;
+using Microsoft.KernelMemory.MemoryStorage;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Microsoft.KernelMemory.MemoryDb.AzureCosmosDBMongoDB;
+
+/// <summary>
+/// Azure CosmosDB Mongo vCore connector for Kernel Memory, read more about Mongo vCore and vector search using Mongo vCore here.
+/// https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/
+/// https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/vector-search
+/// </summary>
+public class AzureCosmosDBMongoDBMemory : IMemoryDb
+{
+    private readonly ITextEmbeddingGenerator _embeddingGenerator;
+    private readonly ILogger<AzureCosmosDBMongoDBMemory> _log;
+    private readonly AzureCosmosDBMongoDBConfig _config;
+    private readonly MongoClient _cosmosDBMongoClient;
+    private readonly IMongoDatabase _cosmosMongoDatabase;
+    private readonly IMongoCollection<AzureCosmosDBMongoDBMemoryRecord> _cosmosMongoCollection;
+    private readonly string _collectionName;
+
+    /// <summary>
+    /// Create a new instance
+    /// </summary>
+    /// <param name="config"> Azure Cosmos DB Mongo vCore configuration</param>
+    /// <param name="embeddingGenerator">Text embedding generator</param>
+    /// <param name="databaseName">Database name for Mongo vCore DB</param>
+    /// <param name="collectionName">Collection name for Mongo vCore DBr</param>
+    /// <param name="log">Application logger</param>
+    public AzureCosmosDBMongoDBMemory(
+        AzureCosmosDBMongoDBConfig config,
+        ITextEmbeddingGenerator embeddingGenerator,
+        string databaseName,
+        string collectionName,
+        ILogger<AzureCosmosDBMongoDBMemory>? log = null)
+    {
+        this._embeddingGenerator = embeddingGenerator;
+        this._log = log ?? DefaultLogger<AzureCosmosDBMongoDBMemory>.Instance;
+        this._config = config;
+
+        if (string.IsNullOrEmpty(config.ConnectionString))
+        {
+            this._log.LogCritical("Azure Cosmos DB Mongo vCore connection string is empty.");
+            throw new AzureCosmosDBMongoDBMemoryException("Azure Cosmos DB Mongo vCore connection string is empty.");
+        }
+
+        if (this._embeddingGenerator == null)
+        {
+            throw new AzureCosmosDBMongoDBMemoryException("Embedding generator not configured");
+        }
+
+        MongoClientSettings settings = MongoClientSettings.FromConnectionString(config.ConnectionString);
```

Comment:
does AzureMongoDb support Managed Identities and `DefaultAzureCredential`?

Reply:
I have added MongoClient to be passes as a variable in the AzureCosmosDBMongoDBMemory constructor. Now if the users want to use AzureKeyCredential, they can use it and directly send the MongoClient to the ctor, or use the vanilla version and pass in only the ConnectionString of the MongoDB cluster in the AzureCosmosDBMongoDBConfig.

Reply:
I'd suggest to take a look at the other Azure connectors, they all support Managed Identity. The approach you suggest, about injecting a custom client, would require custom code for anyone using Prod deployments. 

For instance if you look at the "infra" folder and the "Deploy to Azure" button, it allows to deploy the solution in Prod with Managed Identity without changing the code, simply by setting "AuthType" to "AzureIdentity" in the config file.